ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/attestation/AggregatingAttestationPool.java

```diff
@@ -100,13 +103,15 @@ private void removeFromSlotMappings(final UnsignedLong slot, final Bytes32 dataR
         .filter(group -> group.getAttestationData().canIncludeInBlockAtSlot(slot))
         .flatMap(MatchingDataAttestationGroup::stream)
         .limit(attestations.getMaxSize())
+        .map(ValidateableAttestation::getAttestation)
         .forEach(attestations::add);
     return attestations;
   }
 
   public synchronized Optional<Attestation> createAggregateFor(
       final AttestationData attestationData) {
     return Optional.ofNullable(attestationGroupByDataHash.get(attestationData.hash_tree_root()))
-        .flatMap(attestations -> attestations.stream().findFirst());
+            .flatMap(attestations -> attestations.stream().findFirst())
+            .map(ValidateableAttestation::getAttestation);
```

Comment:
This makes me think that inheritance might be easier to manage ValidateableAttestations than delegation. If we chose inheritance, we would have access to both ValidateableAttestation and normal Attestation of the same attestation instance. Any thoughts on this?

Reply:
Made the change btw.

Reply:
I missed this comment. Inheritance is tempting but I'd avoid it.  Generally preferring delegation over inheritance leads to much better composability and less complexity.

I had considered if we should make `Attestation` an interface and then we'd have `ValidatableAttestation` and `AttestationImpl` (the current class for want of a better name) both implement it so you could still use a `ValidatableAttestation` as an `Attestation`.  But the lack of ability to find a good name is a sign that this isn't a great design and things like SSZ serialisation would likely cause us a lot of grief (another good reason to avoid reflection based magic).